### ISSUE TITLE
Thread about potential improvements for the Foundations of Machine Learning course on JuliaAcademy.com

### ISSUE THREAD

**Issue by logankilpatrick** [JuliaComputing/JuliaAcademyMaterials#24]
Feedback thread created during Google Code-In.

**Comment by soumitradev** [comment]
Chris has done an amazing job explaining the very basics of machine learning and how we approach this impossible-seeming idea of making a machine learn. This course is actually flawless as it is, but it could be even better with a few improvements:
- Maybe we can add a couple of exercises that test the student on commonly used conventions, keywords, and basic concepts?
- The course could use a standard dataset (such as the MNIST dataset) as an example in addition to the general Apples v/s Bananas analogy used in this course so that the students are more prepared later to tackle such real-world ML applications. Maybe the application of these concepts on famous datasets could be an extension to the course?
- The code in the video is Julia 0.6. I think most of the code as shown in the video is the same, but there is a chance that the video using Interact.jl needs updating.

**Comment by avinash010502** [comment]
The tutorial is really interesting. Instructor Jane Herriman has done a wonderful job and it is very easy to understand the topics. I am learning C language in school and I could very easily relate these topics  to those in C. I was looking forward to learn python and I learnt the basics of python using online tutorials.. As python and Julia are similar, at a point I got confused between them but the instructor has given so many examples and so I can clearly find the difference and finally I got it cleared.
I don't think there is anything negative in this tutorial,
[TUTORIAL.pdf](https://github.com/JuliaComputing/JuliaAcademyMaterials/files/4010147/TUTORIAL.pdf)
 its just doing great and if I have to give a comment it can only be a positive compliment. Its really great to teach this in such an easy way. Thanks to Julia and the instructor.

**Comment by avinash010502** [comment]
The tutorial is really interesting. Instructor Jane Herriman has done a wonderful job and it is very easy to understand the topics. I am learning C language in school and I could very easily relate these topics  to those in C. I was looking forward to learn python and I learnt the basics of python using online tutorials.. As python and Julia are similar, at a point I got confused between them but the instructor has given so many examples and so I can clearly find the difference and finally I got it cleared.
I don't think there is anything negative in this tutorial,
[TUTORIAL.pdf](https://github.com/JuliaComputing/JuliaAcademyMaterials/files/4010147/TUTORIAL.pdf)
 its just doing great and if I have to give a comment it can only be a positive compliment. Its really great to teach this in such an easy way. Thanks to Julia and the instructor.
The tutorial is really interesting. Instructor Jane Herriman has done a wonderful job and it is very easy to understand the topics. I am learning C language in school and I could very easily relate these topics  to those in C. I was looking forward to learn python and I learnt the basics of python using online tutorials.. As python and Julia are similar, at a point I got confused between them but the instructor has given so many examples and so I can clearly find the difference and finally I got it cleared.
I don't think there is anything negative in this tutorial, its just doing great and if I have to give a comment it can only be a positive compliment. Its really great to teach this in such an easy way. Thanks to Julia and the instructor.
[TUTORIAL.pdf](https://github.com/JuliaComputing/JuliaAcademyMaterials/files/4010201/TUTORIAL.pdf)

**Comment by sbhavyasivaraj** [comment]
The tutorial is really interesting. Instructor Jane Herriman has done a wonderful job and it is very easy to understand the topics. I am learning C language in school and I could very easily relate these topics  to those in C. I was looking forward to learn python and I learnt the basics of python using online tutorials.. As python and Julia are similar, at a point I got confused between them but the instructor has given so many examples and so I can clearly find the difference and finally I got it cleared.
I don't think there is anything negative in this tutorial, its just doing great and if I have to give a comment it can only be a positive compliment. Its really great to teach this in such an easy way. Thanks to Julia and the instructor.
[TUTORIAL.pdf](https://github.com/JuliaComputing/JuliaAcademyMaterials/files/4010147/TUTORIAL.pdf)

**Comment by soumitradev** [comment]
@sbhavyasivaraj I think you are talking about [this](https://github.com/JuliaComputing/JuliaAcademyMaterials/issues/23) thread, not the Foundations of Machine Learning one.

**Comment by mikolajhojda** [comment]
I think there should be a test in the tutorial. Maybe questions about definitions? I wish that course weren't as YouTube but as in "Introduction to Julia". The examples should be easier for a rookie in machine learning.

**Comment by avinash010502** [comment]
@soumitradev I just realized that I interchanged the tasks and I have given the feedback for the other task, thank you, I will do the needful.

**Comment by sbhavyasivaraj** [comment]
@soumitradev I just realized that i have interchanged the tasks and I have given the feedback for the other one,thank you, i will do the needful

**Comment by josobar** [comment]
I really enjoyed this tutorial. The information is interesting and the graphics and drawings help the viewer to understand the concepts well. I also found it interesting to learn about the instructor not only through the instructor's bio but also through the short "Motivation" video at the beginning of the course. Learning about the instructor's goals and interests helps form a connection between student and teacher that I believe is very important. 

Suggestions:
1. Towards the end of the course, broaden more complex examples of how machine learning can be used other than identifying apples and bananas. 
2. I know that some students prefer to learn through reading rather than videos, so maybe include a transcript of the video tutorials along with some pictures in a pdf?

**Comment by D3nii** [comment]
Okay, tbh the course was pretty fast paced and poorly managed.

**Comment by aftex261** [comment]
So here what I have experienced throughout your course:-
1.The course lacks connections and the proper interactive way of teaching.
2.In Representing data with models,nice explanation has been done in consideration to the apple example.
3.In Building models ,concepts can be made more interesting by focusing on the basics.
4.Rest of the lectures were pretty interesting and the explanation of Sigmoid function has been done very nicely with respect to the variation of values of b and w.
So,At last I would like to say that the overall course was pretty interesting and has been nicely crafted but it could be more better if you follow the above points.
My rating:-4.4/5

**Comment by TheComputerM** [comment]
The course was wonderful and the instructor goes nice and slow for anyone to understand the point he is making, he explained the basics very well. The only thing that was a problem for me was not being able to quickly interact with the graphs and equations he was writing. It would be great if the course viewed the attached .ipynb file at the bottom so we don't have to go through messy setup, Or better provide a link to https://mybinder.org/.

**Comment by TheComputerM** [comment]
And I forgot to mention that adding exercises to this course is very important

**Comment by sbhavyasivaraj** [comment]
Machine learning, in general, is a very tough course to understand. The amount of time spent to
learn it, has to be more, but instructor Chris made it easy and it was not much harder too. At times,
it was difficult to understand the accent of the instructor, but as subtitles was also provided for each
and every video, it was better for understanding. If there should be a point of making this a better
course for children, the examples can be changed into an interesting one and the instructor could be
clearer in giving examples and explaining. There should be a session of asking questions as it was in
the previous ones, this can be done to check if the people have really understood the concepts and
that may tell the academy what they have to improve in. If more number of people have got the
answer right, then it has reached well to people, if not they can make changes on account of it. One
more positive feedback is that we can take our own time to understand the topics. We can go to the
previous lecture and listen again if one thinks that a particular topic is not clear. There can be more
interactive sessions, like the course can be made more interesting because now it is very blunt for
kids of around 13-14 who try GCI for the first time. It can make the kids to try Julia every year for
GCI.
Bhavya S

**Comment by sbhavyasivaraj** [comment]
Machine learning, in general, is a very tough course to understand. The amount of time spent to
learn it, has to be more, but instructor Chris made it easy and it was not much harder too. At times,
it was difficult to understand the accent of the instructor, but as subtitles was also provided for each
and every video, it was better for understanding. If there should be a point of making this a better
course for children, the examples can be changed into an interesting one and the instructor could be
clearer in giving examples and explaining. There should be a session of asking questions as it was in
the previous ones, this can be done to check if the people have really understood the concepts and
that may tell the academy what they have to improve in. If more number of people have got the
answer right, then it has reached well to people, if not they can make changes on account of it. One
more positive feedback is that we can take our own time to understand the topics. We can go to the
previous lecture and listen again if one thinks that a particular topic is not clear. There can be more
interactive sessions, like the course can be made more interesting because now it is very blunt for
kids of around 13-14 who try GCI for the first time. It can make the kids to try Julia every year for
GCI.
Bhavya S

**Comment by chinglamchoi** [comment]
**1. Visual and clear**
I really appreciated how the instructor used plots and interact.jl to illustrate various concepts. His demonstration for adjusting the weights and bias was genius! By visualising the graph of sigmoid, the instructor is able to intuitively explain both the concept of a discriminative decision boundary and gradient descent, which is hugely impressive!

**2. Maths and intuition**
It was nice that the instructor touched on some maths behind the algorithms, but perhaps a bit more elaboration would be helpful. For instance, dot product isn't exactly intuitive but is a crucial concept to understand in ML. While offering the Wikipedia definition of "the product of the Euclidean magnitudes of the two vectors and the cosine of the angle between them" might not be suitable for a foundations course, perhaps he could said something along the lines of "it is useful in determining the similarity (in terms of vector directions) of 2 vectors", which saves a lot of trouble later understanding things like triplet loss and attention mechanisms. Moreover, I think it would be nice to expatiate a bit more on the update rule for gradient descent. Deriving the update rule to the weights would be nice!

**3. Covering more concepts**
The instructor mentioned that Sum of Squares error was nice to work with using Gradient Descent but didn't say why. It might be nice to cover concepts of convexity and convergence. Also, perhaps more discussion on overfitting would have been helpful. Topics like the bias-variance tradeoff (while slightly outdated in the era of big data), size of dataset, train-valid-test split are very interesting!

**Comment by D3nii** [comment]
The only thing I would say that there but be some kind of assignments in the course so that the students can tested of what have they learnt and this will help them to clear any ambiguities they would be having after seeing the videos as well. Furthermore, the course was pretty good and covered the topics but I think there must be a video of the type of Artificial Intelligence, i.e. ML, DL, NN.

**Comment by Hry789** [comment]
First of all, I would express my appreciation for your wonderful job in teaching the Machine learning course.  Your deep knowledge is certainly a big factor for us to succeed in this field.  Today, I took the course,thinking that it will be useful although it currently not a popular language. 

Thank you in advance for your patience teaching us in this aspect.
Best regards for your teaching, will keep you posted on the progress of my exams and job searching.! Take care.

**Comment by CompSciDZ** [comment]
WOW!

I had no idea that the course would completely make something that sounds so difficult easy to visualize and comprehend. The instructor did a superb job of explaining each concept the way a machine would see something we see, and simplify it to the point where it is understandable among others.

One improvement that can be made is the integration of the Jupyter Notebook or any development environment into the videoes to show the user how the code will execute. The course visualizes Machine Learning, but doesn't really go into how to write/code with it. That would be a nice add-on.

The addition of assessments would also be useful for this topic with questions that have to do with the video comprehension:

(Ex. Why can model complexity potentially have a negative effect on Machine Learning? or Scenario Questions: What would the model look like for this case?)

All in all, the course is executed wonderfully and I enjoyed it!

**Comment by TARUN49157** [comment]
The instructor chris made a really good job in explaining the basics of machine learning and made it very easy to understand each of the concept.Though the course is very good,there are a few suggestions :
1.it would be really good if we add a few interactive courses.
2.addition of assignments

**Comment by ishita272001** [comment]
It was a very good course but it has to be made some improvements.
1.The example (apple and banana) being given in video is little bit difficult to understand.
2.When I opened the PDF it contains only codes not the written format of the lecture.Written form of lecture should be added.
3.A review test should be added at last. The test can contain questions related to what is explained in video.
4.The course is running little bit fast so speed of this course should be decreased.

**Comment by kchatr** [comment]
The course was a great introduction to machine learning classification and takes time to go through the fundamentals, as well as implementing the code. Talking about the sigmoid function in-depth, discussing what _w_ and _b_ do was greatly beneficial, and can assist in setting the hyperparameters. However, I found the course more beneficial from a theoretical point-of-view, and understand how classification works and how to adjust the sigmoid, but it went into very little detail about the code implementation, syntax, and functionality. It would be great if there was a line by line explanation of what that line did, as well as what its syntax means. 

Complex topics and algorithms were explained both mathematically and in layman terms, not overly clouding it with buzzwords and obscure terms. The video _What is Learning?_ was an exemplary demonstration of this, and someone who had only experienced linear regression was able to quickly see how classification works and how the algorithms are implemented.

The visuals and graphic style of presentation were greatly beneficial and made the course more engaging and interesting. The graphs shown, along with the code, were quite helpful and made the video feel more down-to-earth, to an extent. 

A bug I found was the inability to pause the videos, which can be greatly frustrating at times and an inconvenience.

**Comment by leotuncer** [comment]
This video was a great introduction to Machine Learning, I thought the use of the sigmoid function and discussing the variables w and b and how they could be used to set the parameters was very fascinating. Although I was unable to view due to the link not looking, I was able to find it on their website.

**Comment by YashasMessi02** [comment]
It was a good introduction to Machine Learning. I could understand the concepts easily and could frame new ideas regarding the concept.

**Comment by Wingfwd644** [comment]
Overall this course was a nice introduction to Machine Learning. It was very informative, and the instructor explained the concepts very well, and it is simple to understand. A few suggestions:1) maybe use more examples, other than bananas and apples? 2) I noticed there is still a lack of practices or interactive quizzes just like the 'introduction to julia' course. Maybe just add a few multiple choice questions? It doesn't need to be much(like only 1 or 2), since the concepts are very easy to grasp.
But yeah, not much to say, good course otherwise.

**Comment by saikishore222** [comment]
entire course was very nice.very good instructor.i was very excited about machine learning.iam very familar with maths which is my favorite subject
my suggestion are:-
1)after every tutorial you have to ask some questions like multiple choice ,based on that user can understand how much he understand.we can test our knowledge .
2)we can add some more topics which is very familar with python.most of the users can have the basic language of python..then the course will have some what more intersting.

**Comment by Hry789** [comment]
This course is great in the fact that you are able to work at your own pace. This can also be the biggest disadvantage. You have to be a very disciplined and organized person to be successful in this course. So I think that mentors should set a schedule to keep track of students.
Besides that,the video should have some comments bar as to let the students to ask their questions just o the spots.Mentors could open a live video to teach us too!
I encourage everyone who has some interest to take this courses. Yes, it's a lot of work and learn, but it is worth it.
Overall the course was great but you must have the discipline required to keep up with the material i was not as good as i should have been but i think i will do good in the end.

**Comment by PriyanshuSahani** [comment]
1. This is indeed a very good course and perhaps, better than the 'Intro to JuliaAcademy' course.
2. Rather than showing complicated sets of codes, the tutorials featured the anatomy of machine learning using a common and understandable language. (This was missing in the 'Intro to JuliaAcademy' course) 
3. The entire tutorial revolved around an example of apple and banana which was again made it easy to understand and was a big plus of the course.
4. The only shortcoming was in the tutorial 'What is Learning?' which certain words were used for some phenomena that were left unexplained and it would have been easier to cope up with that
particular tutorial if they elaborated a little.

**Comment by TechwithSaad** [comment]
I was aware of Machine Learning Technology before but now I know the basics of Machine Learning and how we approach this impossible-seeming idea of making a machine learn. This is indeed a very good course but it could be even better with a few improvements like quizzes, hands on exercises can be added to improve the learning experience of the student.

Thank You!

**Comment by TechwithSaad** [comment]
I was aware of Machine Learning Technology before but now I know the basics of Machine Learning and how we approach this impossible-seeming idea of making a machine learn. This is indeed a very good course but it could be even better with a few improvements like quizzes, hands on exercises can be added to improve the learning experience of the student.

Thank You!

**Comment by schrodincat2718** [comment]
The instructor did a great job and presented the complex machine learning in a simple way. However the course can be even better with slower pace and some exercises provided

**Comment by YashasMessi02** [comment]
It was a good introduction to Machine Learning. I could understand the concepts easily and could frame new ideas regarding the concept.It was very informative and one could easily grasp the concept. But it could've been more interactive like adding quizzes after every lecture to test our knowledge, supplying a document of what we've learnt, and a final video for the summary of all the topics and the topics to be covered in the next part.

**Comment by Redeem-Grimm-Satoshi** [comment]
This is probably the best machine learning course I've taken so far. I think if **exercises are given at each stage and a standard dataset is used other than bananas  and apples** then the learning experience will be awesome.

**Comment by Sam659** [comment]
The course was very informative, especially great as an introduction to Machine Learning. I liked how the teacher used a simple example, like the apple and banana, to explain complex equations. The use of diagrams helped me better understand how a certain equation was affected by a new or changed variable, even though the given equation was above my understanding of mathematics. That being said, I wish that the course was more interactive with the user. For instance, being able to have access to an interactive graph (In the building models video, where one could easily change the b and w value, I wasn’t able to interact with it) , in which the user could play around with in order to better understand the material.

**Comment by TechwithSaad** [comment]
Machine Learning Models are homogeneous to functions that will predict some output for a particular given input.

![20200105_212959](https://user-images.githubusercontent.com/58442432/71786646-64258c80-302f-11ea-83be-8c46852895ca.jpg)
![20200105_213020](https://user-images.githubusercontent.com/58442432/71786647-64258c80-302f-11ea-93ad-ae43822e6bb1.jpg)

You can generate a new model with the same algorithm and different dataset or different algorithm and same dataset to achive the accuracy/best prediction.

Thank You!

**Comment by Ching367436** [comment]
The course is great, and make me understand what is artificial neuron.
I think it's a good start for me.

**Comment by akaashkasi** [comment]
Feedback: I enjoyed learning from the MIT professor: Chris. I was able to understand the majority of the course. However, I wish there was an interactive module along with the course to better understand the material. It would be convenient as well as helpful. I also think that with practice problems that I can possibly do a similar task to the problem that was presented in the course.

**Comment by Redeem-Grimm-Satoshi** [comment]
**Standard datasets**: I mean like datasets used in real world application like the Wine Quality Dataset(involves predicting the quality of white wines on a scale given chemical measures of each wine) is just an example. Something like that would be more preferable.

**Exercises**: Different sort of review questions  and exercises can be given at each stage like asking us to build a model and train it so it solves a particular problem using some datasets given. A leaders board can be created for scores, top learners.

**Comment by Redeem-Grimm-Satoshi** [comment]
This is probably one of the best machine learning course I've taken so far. I think if **exercises** are given at each stage and a **standard datasets** is used other than bananas and apples then the learning experience will be awesome.

**Standard datasets**: I mean like datasets used in real-world applications like the **Wine Quality Dataset**(involves predicting the quality of white wines on a scale given chemical measures of each wine) is just an example. Something like that would be more preferable.

**Exercises**: Different sorts of review questions and exercises can be given at each stage like asking us to build a model and train it so it solves a particular problem using some datasets given. A leader's board can be created for scores, top learners.

**Comment by Pkmnfan02** [comment]
The videos in this course are all under 10 minutes long. This leaves room for oversimplification, not understanding some things and not going in depth with examples.
Having the lectures recorded and not in the class allows you to pause and rewind to rewatch the topic as you wish.

Before I claimed the task, I watched the introduction video to see if it was interesting and I was hooked. I had no idea machine learning involved math, although it seems obvious is me now. The short introduction is good to draw people in.

After the end of each video, a PDF/document of some sorts with a study sheet or review package would beneficial. I found myself starting another video as soon as one was done, a review sheet of what was covered would allow the information to go to long term memory. 

The math equations in the video "What is Learning" and "Introduction to Neurons" was somewhat confusing to me. I haven't taken a lot of math courses yet as its all next semester for me. I was only about to squeeze by (in terms of understanding content) by remembering some things my brother taught me last year

Another quick note: I noticed you can click "complete and continue" to move on to the next video without watching a video all the way through, it could pose some problems in the future if you couldn't go back. But you can switch back and forth between videos easily, so it's good now.

**Comment by srikarthi1144** [comment]
The course was well ordered from motivation to neuron but not elaborate Each module had only few minutes of video lectures which makes viewers to lose interest There was less demo part in the environment and also there where no question and answering part in the tutorial which makes learners unable to test our knowledge But overall even the small concept which mentioned was good and especially the thing about sigmoid activation function was good .He covered the topics sort and sweet and made me to begin with julia

**Comment by NoorHasanShaik86** [comment]
The course is really interesting.The Instructor Dr. Chris Rackauckas has explained the course very well but the course is not elaborate. Each lecture is only a few minutes . These small videos make the students to lose their interest. The lecture only contains the explanation but there are no questions and answers about the lecture or topic, It should be beneficial if you provide that . 
Provide some quizzes about the lecture at the end of the lecture.
Provide some pdfs or documents about the course at the end of the course.
 Finally the course is good.
Thank You.

**Comment by nika02** [comment]
The videos were relatively short and bite-sized which is really good for sustaining interest of the watcher, especially when Machine Learning is a more confusing and complicated topic. 

There could've been more examples in the "Building Models" section so that the watcher can see how it would still apply to another context.

The "What Is Learning?" video did a great job in explaining sum squared error and gradient descent. Another video can be used here to show how that would translate into code.

Personally, the "What are Neurons?" video was slightly confusing to me with the math but the rough general idea is still here.

Short quizzes can be placed after every 2 sections to keep the watcher engaged and to check and apply their understanding after watching the videos.

Overall, the course was easy to follow and understand while still covering the fundamentals of Machine Learning :)

**Comment by qianhuii** [comment]
Here are my feedbacks:
- The small-sized videos are really engaging. The video duration is just nice as an introduction.
- The Jupyter Notebook files are really good for me to improve my understanding by playing around with the variables.
- It would be good to include instructions on how to open .Ipynb files, for example by providing hyperlink to a website with detailed instructions on how to install and use Jupyter Notebook, or how to use Jupyter Notebook online.
- In the lecture "Representing Data with Models", the hyperlink in the word "Notebook" did not work for me. When I clicked on the link "https://staging.juliabox.com/notebook/notebooks/training/aiprohub/iccml/05.ML-Building-models.ipynb", my browser stated "can’t connect to the server at www.staging.juliabox.com".
- In the lecture "Building Models", in the video, it would be better to show the sigmoid formula in a small window on the right (above the lecturer). This is because people like me might not remember exactly how the sigmoid formula looked like. Besides, mathematical inclined people might want to think about how the changes in the variables (w, b) affects the sigmoid graph.
- In the lecture "What are Neurons?", I find it difficult to understand the concept of neurons when it was explained using matrices. Only after playing around with the neuron model then I understand the concept.
Overall, the course is a very good introductory course. I am interested to learn more about machine learning. Thank you for the task.

**Comment by Chenyeeren10** [comment]
I feel that the course is well designed, being short while still able to understand as I had no prior knowledge except for basic coding. Chris Rackauckas has done a good job in explaining the concepts of models, learning and cost equations in a way that is easy to digest. The short video lectures also allowed me to view them separately instead of being one long video, which gave me time to review the information. The notebooks were helpful as it made the course more interactive and hands-on. My feedback is to allow us to use the notebook examples to other contexts through activities so that we can use the code ourselves.

**Comment by dhruv2604-create** [comment]
Hi everyone! I recently completed the "Foundations of Machine Learning" Course on Julia Academy. I believe this is the best course on Julia Academy. Although there are some suggestions that I would like to make:-

The audio should definitely be improved.
There should be comment sections where users can complement as well as ask questions based on the lectures and the lecturers can answer. Also, certain live QnAs should be held with users to further develop knowledge of users.
New courses should be made aligned with new versions of Julia.
There should be a feature for users to practice alongside listening to lectures on the site itself.
Tests should be taken after a user completes a certain component of the course and he or she should only be able to advance after successfully completing the test.

**Comment by kgauld1** [comment]
Hi all,
I just finished the "Foundations of Machine Learning" Course on Julia Academy.  It was a really great learning experience, as I am very interested in machine learning.  Possibly my only suggestion for this course is to make it a bit longer- It seems to end just where the whole "learning" idea begins, which may turn off some who are less experienced in machine learning.  Overall a good course though!

**Comment by amyhu0114** [comment]
The course is really helpful for beginners as it incorporates examples that are easy to understand. I enjoy learning the basics of machine learning through short videos.
Here are my suggestions for potential improvements:
Maybe include some hands-on exercises where students could try out the concepts learned themselves
Include some quizzes with multiple choice/short answer questions to ensure the students understand the concepts
The course is great overall! =)

**Comment by write2taran** [comment]
I found the course was great and even the language was easy to understand. How ever for some people who may be unable to understand spoken English clearly due to it being a second language for them, it's a bit difficult to under stand. The auto generated subtitles an't perfect' there's a need to add subtitles. Other wise, instructor(sir) has done a great job

**Comment by Akshat-mehrotra** [comment]
This course failed to cover any other functions except sigmoid and MSE. Which I believe to be its biggest weakness. Lack of mathematical elaboration on functions and the lack of explanation about the whys of choosing said functions and operations is also what brings down the overall score. For example, unlike the loss function discussed in the video, the reason for the sigmoid function to be the way it is and its pros and cons are not discussed. As the course is uploaded on the site JuliaAcademy I expected a bit more Julia code to be in the videos.

Otherwise, the video is paced at a good speed, and everything which is explained can be grasped easily. My go-to site for any machine learning tutorial is [machinelearningmastery.com](https://machinelearningmastery.com/). Its explanations of the Convolution networks is unparalleled

**Comment by amanjincodeout** [comment]
BEST Ever course which gives insight of real foundations of machine learning.

**Comment by CataniciuJasmine** [comment]
The videos presented in the learning environments are not very well presented. Some have to be tracked twice to understand.
It would be better if after each video there was a questionnaire with two three questions easily structured so that we can understand better for the future.

Google code-in
[
![all j a](https://user-images.githubusercontent.com/60110542/72747738-280d3280-3bbe-11ea-9216-db384a29339f.jpg)
](url)

**Comment by priyansh0512** [comment]
Chris has done a great job in making us understand what machine learning is.

**Comment by priyansh0512** [comment]
Chris has done a great job in making us understand what machine learning is. The way in which he explains the course is absolutely amazing and according to me should be understandable by all. He took great examples of Apple and Banana to explain us and get our concepts clear.

**Comment by priyansh0512** [comment]
An improvement could be in the captions part of the video if the learner is not fluent in understanding english.

**Comment by tomahu111** [comment]
Explanaitions, topics and explaining what are neurons (what many courses dont do) are good things about the course, however there is a lot of stuff that could be done better.
First problem to see are the files. you download, there culd be at least link to guide how to install jupyter with julia or, if you want, instead of files there could be links to google coolab like in tensorflows course. Another thing is that despite good videos there is not perceptible connection between videos and those articles/code examples. Thinking about this those are just videos and articles with code examples to download. And last but not least it is really short, you could go beyond talking about apples and bananas.

**Comment by kodingwizard** [comment]
The course was very detailed in explaining the different functions in Machine Learning. A positive point is that the instructor drew and wrote everything he was explaining on his screen to give students a visual representation of the functions. However, there are a few things that could improve:

1) There should be exercises and quizzes given to test the students' understanding of the topic being explained. The assignments could include multiple choice questions, or have the students play around with the mathematical model to match the values of w and b.

2) The course should include a video that gives a real - life example of a machine recognizing an object using the functions described and the features of the object. The students would probably have a better understanding of what Machine Learning is if they saw its use and the way the computer uses the features to recognize an object.
